bus_frequency/src/gtfs.rs

```diff
@@ -0,0 +1,151 @@
+use std::collections::BTreeMap;
+
+use anyhow::Result;
+use chrono::NaiveTime;
+use fs_err::File;
+use serde::Deserialize;
+
+// See https://gtfs.org/schedule/reference/. This groups all arrival_times per day and stop, with
+// some caveats:
+//
+// - Ignoring calendar_dates.txt, which lists days an extra service is run or is skipped
+// - Ignoring the start/end date ranges in calendar.txt
+// - Ignoring times after midnight (unlikely to be the peak hour)
+// - Ignoring frequencies.txt, which is used for only a few services in the UK GTFS
+// - Skipping some services that don't map to trips
+
+pub struct Stop {
+    pub name: String,
+    pub lon: f64,
+    pub lat: f64,
+
+    // Indexed by Day
+    pub times_per_day: Vec<Vec<NaiveTime>>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum Day {
+    Monday = 0,
+    Tuesday = 1,
+    Wednesday = 2,
+    Thursday = 3,
+    Friday = 4,
+    Saturday = 5,
+    Sunday = 6,
+}
+
+/// Takes a path to a GTFS directory, and returns stop ID mapped to a Stop
+pub fn get_stops(dir_path: &str) -> Result<BTreeMap<String, Stop>> {
+    println!("Scraping trips.txt");
+    let mut trip_to_service = BTreeMap::new();
+    for rec in csv::Reader::from_reader(File::open(format!("{dir_path}/trips.txt"))?).deserialize()
+    {
+        let rec: TripRow = rec?;
+        trip_to_service.insert(rec.trip_id, rec.service_id);
+    }
+
+    println!("Scraping calendar.txt");
+    let mut service_to_days: BTreeMap<String, Vec<Day>> = BTreeMap::new();
```

Comment:
I'm being a bit loose here; if this was a larger script, wrapper types to distinguish ServiceID, TripID, etc would be useful to prevent silly bugs and for documentation